lms/djangoapps/instructor/tests/test_certificates.py

```diff
@@ -351,69 +352,90 @@ def test_certificate_regeneration_success(self):
             Regenerate certificates for `verified_users_with_audit_certs` is run,
-            Modified certificate status is `verified` if user is ID verified otherwise `unverified`.
+            Modified certificate status is `generating` if user is ID verified otherwise `unverified`.
         """
         # Check that user is enrolled in audit mode.
         enrollment = CourseEnrollment.get_enrollment(self.user, self.course.id)
-        self.assertEqual(enrollment.mode, CourseMode.AUDIT)
-
-        # Generate certificate for user and check that user has a audit passing certificate.
-        with patch('student.models.CourseEnrollment.refund_cutoff_date') as cutoff_date:
-            cutoff_date.return_value = datetime.now(pytz.UTC) - timedelta(minutes=5)
-            cert_status = certs_api.generate_user_certificates(student=self.user, course_key=self.course.id, course=self.course)
-            self.assertEqual(cert_status, CertificateStatuses.audit_passing)
-
-        # Update user enrollment mode to verified mode.
-        enrollment.update_enrollment(mode='verified')
-        self.assertEqual(enrollment.mode, CourseMode.VERIFIED)
-
-        with patch(
-            'lms.djangoapps.verify_student.models.SoftwareSecurePhotoVerification.user_is_verified'
-        ) as user_verify:
-            user_verify.return_value = user_verified
-
-            # Login the client and access the url with 'certificate_statuses'
-            self.client.login(username=self.global_staff.username, password='test')
-            url = reverse('start_certificate_regeneration', kwargs={'course_id': unicode(self.course.id)})
-            response = self.client.post(url, data={'certificate_statuses': ['verified_users_with_audit_certs']})
-
-            # Assert 200 status code in response
-            self.assertEqual(response.status_code, 200)
-            res_json = json.loads(response.content)
-
-            # Assert request is successful
-            self.assertTrue(res_json['success'])
-
-            # Assert success message
-            self.assertEqual(
-                res_json['message'],
-                u'Certificate regeneration task has been started. You can view the status of the generation task in '
-                u'the "Pending Tasks" section.'
-            )
-            # Check user has a not audit passing certificate now.
-            cert = certs_api.get_certificate_for_user(self.user.username, self.course.id)
-            self.assertNotEqual(cert['status'], CertificateStatuses.audit_passing)
-            self.assertEqual(cert['status'], expected_cert_status)
+        self.assertEquals(enrollment.mode, CourseMode.AUDIT)
+
+        with mock_passing_grade():
+
+            # Generate certificate for user and check that user has a audit passing certificate.
+            with mock.patch('student.models.CourseEnrollment.refund_cutoff_date') as cutoff_date:
+                cutoff_date.return_value = datetime.now(pytz.UTC) - timedelta(minutes=5)
```

Comment:
Is this mock really needed as there is a setting override `@override_settings(AUDIT_CERT_CUTOFF_DATE=datetime.now(pytz.UTC) - timedelta(days=1)`  as well. 

I think, I had added these but can you please check if both `mock.patch` and `override_settings` are needed here.